### PR TITLE
Add test for user story 22

### PR DESCRIPTION
**Pull request by meghanstovall**
Complete story 22

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -30,11 +29,15 @@ def edit
...
-    flash[:sucess] = "Changes Made to Profile Successfully"
-    redirect_to '/profile'
+    user.update(user_params)
+    if user.save
+      flash[:sucess] = "Changes Made to Profile Successfully"
+      redirect_to '/profile'
+    else
+      flash[:error] = "#{user.errors.full_messages.to_sentence}"
+      redirect_to "/profile/#{user.id}/edit"
+    end
```
Comment (eamouzou) [inline app/controllers/users_controller.rb:40]: conditional to check for completed fields and lack of duplication emails

File: app/views/items/index.html.erb [hunk 2]
```diff
@@ -28,7 +28,7 @@
       <h2> <%=link_to item.name, "/items/#{item.id}" %> </h2>
       <p>Sold by: <%=link_to item.merchant.name, "/merchants/#{item.merchant.id}" %></p>
       <a href="/items/<%= item.id %>">
-         <img alt="<%= item.name %>" src= <%= item.image %> width="500" height="500">
+         <img alt="<%= item.id %>" src= <%= item.image %> width="500" height="500">
```
Comment (eamouzou) [inline app/views/items/index.html.erb:31]: Changed the alt for image link to avoid duplicate error.

File: db/migrate/20200219224408_add_role_to_users.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 class AddRoleToUsers < ActiveRecord::Migration[5.1]
   def change
-    add_column :users, :role, :integer, default: 0
+    add_column :users, :role, :integer, default: 1
```
Comment (eamouzou) [inline db/migrate/20200219224408_add_role_to_users.rb:3]: Default user changed from visitor to default. If a user is created, it should be the default user which is role 1 for us.

File: spec/features/users/edit_spec.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    expect(current_path).to eq("/profile")
+    click_link "Edit Profile"
+
+    fill_in :email, with: @user_2.email
+    fill_in :password, with: @user_1.password
+    fill_in :password_confirmation, with: @user_1.password_confirmation
+    click_button "Submit"
+
+    expect(current_path).to eq("/profile/#{@user_1.id}/edit")
+    expect(page).to have_content("Email has already been taken")
```
Comment (eamouzou) [inline spec/features/users/edit_spec.rb:45]: add test for error when using a duplicate email
